dvc/output.py

```diff
@@ -474,7 +480,13 @@ def _parse_path(self, fs, fs_path):
             #
             # FIXME: if we have Windows path containing / or posix one with \
             # then we have #2059 bug and can't really handle that.
-            fs_path = fs.path.join(self.stage.wdir, fs_path)
+            wdir = self.stage.wdir
+            if isinstance(fs, GitFileSystem):
+                fs_path = as_posix(fs_path)
+                # dvc always generates wdir as posixpath but converting here
+                # just to be safe if it was written manually
+                wdir = as_posix(wdir)
```

Comment:
I can make this a bit weaker by only calling `as_posix` on Windows. But we are using this kind of translations liberally everywhere where we use dvcfs.

Reply:
We need to convert to `as_posix` at one point anyway.